### ISSUE TITLE
Does KubeVirt play nice with Constellation (confidential computing)?

### ISSUE THREAD

**Issue by jt0dd** [kubevirt/kubevirt#8870]
See this post that I just made on LinkedIn (pasted below): https://www.linkedin.com/feed/update/urn:li:activity:7003044816477409280/

I reference KubeVirt as a way to achieve fully cloud-agnostic infrastrcture-as-code, but it occurs to me that KubeVirt may not play nice with Constellation: https://github.com/edgelesssys/constellation

"Constellation is a Kubernetes engine that aims to provide the best possible data security. It wraps your K8s cluster into a single confidential context that is shielded from the underlying cloud infrastructure. Everything inside is always encrypted, including at runtime in memory."

If it doesn't, I would definitely see that kind of support or at least a confidential computing option for KubeVirt as a huge plus for the K8s world.

---

Let's talk about something fun: cloud-agnostic Compliance-as-Code (CaC).

CaC is not really here yet, but: I believe it is the future of efficient, low-overhead cybersecurity policy enforcement. I'll make a few arguments as to why you should care.

TL;DR: In 2020 Azure published means of automated (during CI/CD) policy enforcement (see image & whitepaper below) and it would benefit every company to try very hard to implement this at the Kubernetes level - not as a cloud native feature through Azure, AWS, GCP, or any other vendor platform.

Because 💸💸💸 .

Automation, Complexity Reduction, & Competition can drastically reduce costs for all businesses in the Infrastructure & Security cost-centers. Here's how:

1️⃣ Automation: Right now, we typically think of Infrastructure-as-Code (IaC) as a way to host applications. But it's certainly possible to use it for orchestration (and as we'll discuss, security auditing) of operational workforce infrastructure too.

2️⃣ Automation: The image below lays out an already-feasible approach through Azure to achieve security policy auditing as-code. Great start to what I envision as a popular cost-saving idea: Compliance-as-Code. Don't fall for it. Yet.

🔹Competition: Being locked into a specific cloud vendor? 💸💸💸 Expensive! The easier it is for you to seamlessly migrate to an alternative cloud platform (or on-prem), the more effectively you and all your peers can drive down cloud costs through the natural process of competition in the cloud computing space. ☁ Cloud agnosticism anyone?

🔹Competition: Kubernetes + Terraform to the rescue ⛑⛑⛑. As much as Terraform claims to be cloud-agnostic, it is not in any comprehensive sense of the word. But add Kubernetes to the mix and it becomes so. A platform-within-a-platform. For anyone pointing out that Kubernetes is "just a container thing", I point you to the KubeVirt plugin, which adds VM support 😏.

🔹 Complexity: Understanding and building expertise on securing 3 different cloud vendors (according to SANS Institute, most companies using cloud use all three major platforms) is hard. So you're going to pay a premium for the skillset, more mistakes will be made (expertise & focus spread thin), and it will take more time. 💸💸💸 Expensive!

3️⃣ So finally, with Kubernetes, Terraform, & KubeVirt in mind, I point you to confidential computing not at the vendor-feature-set level, but at the Kubernetes level: Constellation (link below). Constellation lets you add confidential computing to Kubernetes without relying on a cloud-vendor-specific feature.

4️⃣ Anyone know of a good Kubernetes-level security auditing plugin / toolset? Because if we can throw that into the mix rather than using a vendor-specific feature-set like the one offered by Azure, we can start working toward a world of cloud-agnostic, automated, orchestrated infrastructure & software security policy auditing, with highly competitive (lower cost) cloud markets.
